tests/test_tpm2_save_load_state_2_block

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+if [ "$(id -u)" -ne 0 ]; then
+	echo "Need to be root to run this test."
+	exit 77
+fi
+
+STATEDIR=$(mktemp -d)
+STATEIMG=$STATEDIR/tpm2.img
+STATEFILE=""
+
+trap "cleanup" SIGTERM EXIT
+function cleanup()
+{
+	rm -rf $STATEDIR
+	if [ -n "$STATEFILE" ]; then
+		losetup -d $STATEFILE
+	fi
+}
+
+# allocate 4 MiB file
+fallocate -x -l $((4 * 1024 * 1024)) "$STATEIMG"
+# and loop mount it
+STATEFILE=$(losetup --show -f $STATEIMG)
```

Comment:
Can this not work on plain files? Using files would not require root rights.

Reply:
.... otherwise it could work with loopback mounted files when root rights are available, with plain files otherwise, or both cases and skipping the loopback testing when no root rights are available ?

Reply:
`test_tpm2_save_load_state_2_block` tests block devices via loopdev (or skips if root is unavailable) and `test_tpm2_save_load_state_2_file` tests on regular files, even without root. This way, both code paths are exercised.